develop/tutorials/articles/liferay-forms/01-ddm-io-api.markdown

```diff
@@ -0,0 +1,155 @@
+    `DDMFormSerializerSerializeRequest` and returns a
+    `DDMFormSerializerSerializeResponse`  with the serialized string in it.
+
+        @Override public DDMFormSerializerSerializeResponse serialize(
+        DDMFormSerializerSerializeRequest ddmFormSerializerSerializeRequest) {
+
+                DDMForm ddmForm = ddmFormSerializerSerializeRequest.getDDMForm(); 
+
+                ...YOUR CODE FOR BUILDING A YAML OBJECT GOES HERE ...  
+
+                DDMFormSerializerSerializeResponse.Builder builder = 
+                    DDMFormSerializerSerializeResponse.Builder.newBuilder(yamlObject.toString());
+
+                return builder.build(); }
+
+This is what you need to create your serializer. Of course, _YOUR CODE FOR
+BUILDING A YAML OBJECT GOES HERE_ requires some explanation. While you can
+do whatever you want here, there are several things you really ought to
+do:
+
+**Add the available Language IDs:** Since you have the `DDMForm` object from the
+    request, call `ddmForm.getAvailableLocales()`.
+
+**Add the default Language ID:** Get this from the `DDMForm` object by calling
+    `ddmForm.getDefaultLocale()`.
+
+**Add the Form Fields:** Get these from the `DDMForm` object by calling
+    `ddmForm.getDDMFormFields()`.
+
+**Add any Form Rules:** Get them form the `DDMForm` object with
+    `ddmForm.getDDMFormRules()`.
+
+**Add Success Page Settings:** Get these from the `DDMForm` with
+    `ddmForm.getDDMFormSuccessPageSettings()`.
+
+All these are done in the default form serializer, `DDMFormJSONSerializer`.
+
+If you have the @product@ source code, you can find the default serializer in
+
+    modules/apps/dynamic-data-mapping/dynamic-data-mapping-io/src/main/java/com/liferay/dynamic/data/mapping/io/internal/DDMFormJSONSerializer.java
+
+You didn't create serialization code for no reason. You'll want to call it from
+somewhere.
+
+## Calling the Serializer 
+
+To use the serializer:
+
+1. Get the serializer from the `DDMFormSerializerTracker`.
+
+2.  Pass the `DDMFormSerializerSerializeRequest` to the tracker.
+
+3.  Receive the `DDMFormSerializerSerializeResponse` with the serialized value.
+
+Here's a code example:
+
+    DDMFormSerializer ddmFormSerializer =
+    ddmFormSerializerTracker.getDDMFormSerializer("yaml");
+
+    DDMFormSerializerSerializeRequest.Builder builder =
+    DDMFormSerializerSerializeRequest.Builder.newBuilder(ddmForm);
+
+    DDMFormSerializerSerializeResponse ddmFormSerializerSerializeResponse =
+    ddmFormSerializer.serialize(builder.build());
+
+    return ddmFormSerializerSerializeResponse.getContent();
+
+In addition to reviewing what I have here, there are some additional questions:
+
+1.  When does this serialization code get called? After a form save?
+2.  Is there configuration in the forms UI to control which serializer is used? Or is it entirely backend code?
+3.  If it's all backend, are developers expected to do something so that their serializer replaces the default JSON serializer?
```

Comment:
When doing things only in the backend the developer should replace the type of the storage type to the desired format.